app/routes.py

```diff
@@ -1,2 +1,190 @@
+    one_task = Task.query.get(task_id)
+    if one_task is None:
+        return jsonify(one_task), 404
+    one_task.completed_at = None
+    db.session.commit()
+    return jsonify({"task": one_task.task_dict()}), 200
+
+
+@goal_bp.route("", methods=["POST"])
+def post_goal():
+    """ Creates new instance of Goal class """
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+    new_goal = Goal(
+        title=request_body["title"]
+    )
+    db.session.add(new_goal)
+    db.session.commit()
+    return jsonify({"goal": new_goal.goal_dict()}), 201
+
+
+@goal_bp.route("", methods=["GET"])
+def get_goal():
+    """ Uses http GET method to access all goal instances in database """
+    goals = Goal.query.all()
+    goals_response = [goal.goal_dict() for goal in goals]
+    return jsonify(goals_response), 200
+
+
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    """ Uses http GET method to access one specific instance of goal class based on goal id """
+    one_goal = Goal.query.get(goal_id)
+    if one_goal is None:
+        return jsonify(one_goal), 404
+    return jsonify({"goal": one_goal.goal_dict()}), 200
+
+
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def put_one_goal(goal_id):
+    """ Uses http method PUT to edit information in specific instance of goal class """
+    one_goal = Goal.query.get(goal_id)
+    if one_goal is None:
+        return jsonify(one_goal), 404
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+    one_goal.title = request_body["title"]
+    db.session.commit()
+    return jsonify({"goal": one_goal.goal_dict()}), 200
+
+
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_one_goal(goal_id):
+    """ Uses http DELETE method to allow user to delete specific instance of goal class """
+    one_goal = Goal.query.get(goal_id)
+    if one_goal is None:
+        return jsonify(one_goal), 404
+    db.session.delete(one_goal)
+    db.session.commit()
+    return jsonify({"details": f"Goal {one_goal.goal_id} \"{one_goal.title}\" successfully deleted"})
+
+
+@goal_bp.route("/<goal_id>/tasks", methods=["POST"])
+def post_goal_tasks(goal_id):
+    """ Uses http POST method to add multiple task instances to one specific goal instance """
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return jsonify(goal), 404
+    request_body = request.get_json()
+    for num in request_body["task_ids"]:
+        task = Task.query.get(num)
+        task.goal_id = goal_id
+        goal.tasks.append(task)
+    db.session.commit()
+    return jsonify({"id": goal.goal_id, "task_ids":
+                    [task["id"] for task in goal.goal_task_dict()["tasks"]]}), 200
+
+
+@goal_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_goal_tasks(goal_id):
```

Comment:
👍  Great work Rebecca, your helper method really helped in cleaning up this logic.